clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiConverter.groovy

```diff
@@ -553,11 +553,13 @@ class KubernetesClientApiConverter {
       case KubernetesVolumeSourceType.Secret:
         def res = new V1SecretVolumeSource()
         res.secretName = volumeSource.secret.secretName
-        volume.secret = res.secretName
+        volume.secret = res
         break
 
       case KubernetesVolumeSourceType.ConfigMap:
         def res = new V1ConfigMapVolumeSource()
+        res.name = volumeSource.configMap.configMapName
```

Comment:
Is it possible that configMap is null here?

Reply:
The validation logic will error out.

Exception ( Create Server Group )
deployKubernetesAtomicOperationDescription.volumeSources[0].configMap.configMapName.empty